exporter/datadogexporter/factory.go

```diff
@@ -100,3 +104,32 @@ func createMetricsExporter(
 		exporterhelper.WithRetry(exporterhelper.CreateDefaultRetrySettings()),
 	)
 }
+
+// createTraceExporter creates a trace exporter based on this config.
+func createTraceExporter(
+	_ context.Context,
+	params component.ExporterCreateParams,
+	c configmodels.Exporter,
+) (component.TraceExporter, error) {
+	// TODO review if trace export can be supported on Windows
+	if runtime.GOOS == "windows" {
+		return nil, errors.New("datadog trace export is currently not supported on Windows")
+	}
+
+	cfg := c.(*Config)
+
+	params.Logger.Info("sanitizing Datadog metrics exporter configuration")
+	if err := cfg.Sanitize(); err != nil {
+		return nil, err
+	}
+
+	exp, err := newTraceExporter(params.Logger, cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	return exporterhelper.NewTraceExporter(
```

Comment:
This sounds like a good solution to me *if* we can avoid the check when we only have the metrics part of the exporter enabled, otherwise I would just go with a loud warning

Reply:
@jpkrohling I agree that this is a good approach however don't really see a way to check for processors on the `component.Host` as far as i can tell, https://github.com/open-telemetry/opentelemetry-collector/blob/96767632d2208306de42bd3e5a531114e4b7cf02/component/component.go#L61-L94 . 

So I think we'd be blocked on adding the `Start` method discussed due to that that limitation in core. 

Broadly speaking i think we're Ok without a hard requirement on a 10s batch processor being set, for now. I agree with your comments here, https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/1266#discussion_r508344022 that this is not a wonderful user experience, but it doesn't necessarily break anything if not followed, it just leads to potential metric inaccuracy for some metrics related to the traces. The traces themselves are fine. We'd ideally like to just get this in the hands of users and see their feedback before overthinking the setup restrictions/details too much.

Reply:
Sounds good to me.